src/app-logic/web-channel.js

```diff
@@ -163,6 +259,25 @@ function _sendMessageWithResponse<Returns: MessageFromBrowser>(
 
     window.addEventListener('WebChannelMessageToContent', listener, true);
 
+    // Add the requestId to the message. The (...: any) is due to
+    // https://github.com/facebook/flow/issues/1488 which is probably fixed
+    // in flow v0.111.
```

Comment:
This is probably not a problem if you also use the spread operator for the type definition... but in light of our possible migration to typescript it's better to keep the intersection like you did. Well I'll leave up to you :-)

Reply:
I believe I tried that. Spreading on unions is just completely busted prior to 111. See "Spreads no longer conflate unions into a single object" on https://medium.com/flow-type/spreads-common-errors-fixes-9701012e9d58